python_package/madflow/custom_op_generator.py

```diff
@@ -0,0 +1,165 @@
+        for s in signatures_:
+            signatures.append(s)
+        signature_variables = []
+        for s in signature_variables_:
+            signature_variables.append(s)
+        function_list = []
+        for f in function_list_:
+            function_list.append(f)
+        headers = []
+        for h in headers_:
+            headers.append(h)
+        headers.append("matrix_" + process_name + ".h")
+
+        custom_op_list = []
+
+        signatures, signature_variables = read_signatures(signatures, signature_variables, _file_)
+
+        signature_variables = convert_signatures(signatures, signature_variables)
+
+        function_list = read_file_from_source(
+            function_list, _file_, signatures, signature_variables
+        )
+
+        matrix_name = "matrix_1_" + process_name + ".py"
+
+        signatures, signature_variables = read_signatures(
+            signatures, signature_variables, matrix_source
+        )
+        signature_variables = convert_signatures(signatures, signature_variables)
+
+        function_list = extract_matrix_from_file(
+            function_list, matrix_source, signatures, signature_variables
+        )
+
+        for i in range(len(function_list)):
+            function_list = check_variables(i, function_list)
+
+        for i in range(len(function_list)):
+            function_list = check_lines(i, function_list)
+        for i in range(len(function_list)):
+            function_list = check_variables(i, function_list)
+
+        function_list[-1] = serialize_function(function_list[-1])
+
+        custom_op_list.append(define_custom_op(function_list[-1]))
+
+        function_list[-1], constants = extract_constants(function_list[-1], constants)
+
+        function_list[-1] = remove_real_ret(function_list[-1])
+
+        # write the Op for both CPU and GPU
+        for device in devices:
+            write_custom_op(
+                headers,
+                namespace,
+                defined,
+                constants,
+                cpuConstants,
+                function_list,
+                custom_op_list,
+                destination,
+                process_name,
+                device,
+            )
+
+        # write matrix_xxxxx.h
+        temp = ""
+        for c in custom_op_list:
+            temp += write_header_file(c, function_list[-1])
+        with open(destination + "gpu/matrix_" + process_name + ".h", "w") as fh:
+            fh.write(temp)
+
+        # write matrix_1_xxxxx.py
+        temp = ""
+        temp = modify_matrix(matrix_source, process_name, destination)
+        with open(destination + matrix_name, "w") as fh:
+            fh.write(temp)
+
+        # --------------------------------------------------------------------------------------
+
+
+def compile(destination):
+    """Compiles the Custom Operator
+    destination: directory of madflow output"""
+    subprocess.check_output(["/bin/sh", "-c", "cd " + destination + "; make"])
```

Comment:
```suggestion
    subprocess.run("make", cwd=destination, check=True)
```
since you are not using the output this should work just the same. If not, you might need to add `shell=True` See [docs](https://docs.python.org/3/library/subprocess.html#subprocess.run)

Reply:
Done.